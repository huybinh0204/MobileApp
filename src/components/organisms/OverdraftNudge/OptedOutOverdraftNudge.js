import { useNavigation } from '@react-navigation/native';
import React from 'react';
import { Nudge } from '_components/molecules';
import { ICONS, NAVIGATION } from '_constants';
import strings from '_localization/index';
import { HeaderDescription, HeaderTitle } from './OptedOutOverdraftNudge.styles';

const OptedOutOverdraftNudge = ({ testID }) => {
  const { navigate } = useNavigation();

  const navigateToOverdraftSellPage = () => {
    navigate(NAVIGATION.shared.overdraftSell);
  };

  const Header = (
    <>
      <HeaderTitle>{strings.overdraft.nudge.optedOut.header.title}</HeaderTitle>
      <HeaderDescription>{strings.overdraft.nudge.optedOut.header.description}</HeaderDescription>
    </>
  );

  return (
    <Nudge
      icon={ICONS.overdraftProtection}
      buttonText={strings.overdraft.nudge.optedOut.buttonText}
      headerComponent={Header}
      onButtonPress={navigateToOverdraftSellPage}
      testID={testID}
    />
  );
};

export default OptedOutOverdraftNudge;
