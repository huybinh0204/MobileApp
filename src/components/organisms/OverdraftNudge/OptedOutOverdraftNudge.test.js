import { useNavigation } from '@react-navigation/native';
import { fireEvent, render, waitFor } from '@testing-library/react-native';
import React from 'react';
import { NAVIGATION } from '_constants';
import strings from '_localization';
import ComponentWithProviders from '_test/mocks/ComponentWithProviders';
import { navigationMock } from '_test/mocks/navigation';
import OptedOutOverdraftNudge from './OptedOutOverdraftNudge';

jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useNavigation: jest.fn(),
}));

describe('OptedOut Overdraft Protection Nudge', () => {
  const mockedNavigation = navigationMock(jest.fn);

  beforeEach(() => {
    useNavigation.mockReturnValue(mockedNavigation);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should match the snapshot', () => {
    const { toJSON } = render(<ComponentWithProviders component={OptedOutOverdraftNudge} />);

    expect(toJSON()).toMatchSnapshot();
  });

  it('should navigate to overdraft legal agreement when button is clicked', async () => {
    const { getByText } = render(
      <ComponentWithProviders
        component={OptedOutOverdraftNudge}
        componentProps={{ testID: 'OverdraftProtectionNudge' }}
      />
    );

    const overdraftProtectionButton = getByText(strings.overdraft.nudge.optedOut.buttonText);

    await waitFor(() => {
      fireEvent.press(overdraftProtectionButton);
    });

    expect(mockedNavigation.navigate).toHaveBeenCalledWith(NAVIGATION.shared.overdraftSell);
  });
});
