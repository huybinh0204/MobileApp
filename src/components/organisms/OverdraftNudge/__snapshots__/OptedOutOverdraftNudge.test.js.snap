// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OptedOut Overdraft Protection Nudge should match the snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgb(242, 242, 242)",
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <RNSScreenContainer
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          activityState={2}
          collapsable={false}
          forwardedRef={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransition={[Function]}
            pointerEvents="box-none"
            style={
              Array [
                Object {
                  "display": "flex",
                  "overflow": undefined,
                },
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
            transitionSpec={
              Object {
                "close": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                Object {
                  "flex": 1,
                  "marginTop": 0,
                }
              }
            >
              <View
                collapsable={false}
                forwardedRef={[Function]}
                needsOffscreenAlphaCompositing={false}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "alignSelf": "stretch",
                              "backgroundColor": "#FFFFFF",
                              "borderColor": "#F2F3F4",
                              "borderRadius": 10,
                              "borderWidth": 0,
                              "padding": 29,
                              "shadowColor": "#362E60",
                              "shadowOffset": Object {
                                "height": 7,
                                "width": 0,
                              },
                              "shadowOpacity": 0.1,
                              "shadowRadius": 10,
                            },
                            Array [
                              Object {
                                "marginBottom": 29,
                              },
                              undefined,
                            ],
                          ]
                        }
                      >
                        <View
                          style={
                            Array [
                              Object {
                                "alignItems": "center",
                                "flexDirection": "row",
                                "marginBottom": 29,
                              },
                              undefined,
                            ]
                          }
                        >
                          <SvgMock
                            height={116}
                            width={116}
                          />
                          <View
                            style={
                              Array [
                                Object {
                                  "flex": 1,
                                  "paddingLeft": 29,
                                },
                                undefined,
                              ]
                            }
                          >
                            <Text
                              allowFontScaling={false}
                              style={
                                Array [
                                  Object {
                                    "color": "#1F1D25",
                                    "fontFamily": "Poppins-SemiBold",
                                    "fontSize": 25.5,
                                    "lineHeight": 35.5,
                                    "marginBottom": 3.5,
                                  },
                                  undefined,
                                ]
                              }
                            >
                              Kinly Overdraft
                            </Text>
                            <Text
                              allowFontScaling={false}
                              style={
                                Array [
                                  Object {
                                    "color": "#5F6A77",
                                    "fontFamily": "Poppins",
                                    "fontSize": 21.5,
                                    "lineHeight": 30.5,
                                  },
                                  undefined,
                                ]
                              }
                            >
                              Set up direct deposit and receive up
to $100 of fee-free overdraft protection!
                            </Text>
                          </View>
                        </View>
                        <View
                          accessibilityState={
                            Object {
                              "disabled": false,
                            }
                          }
                          accessible={true}
                          collapsable={false}
                          focusable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "alignSelf": "stretch",
                              "backgroundColor": "#EBE9FD",
                              "borderRadius": 10,
                              "opacity": 1,
                              "paddingHorizontal": 45.5,
                              "paddingVertical": 31,
                            }
                          }
                        >
                          <Text
                            allowFontScaling={false}
                            disabled={false}
                            style={
                              Array [
                                Object {
                                  "color": "#3D3397",
                                  "fontFamily": "Poppins-SemiBold",
                                  "fontSize": 25.5,
                                  "lineHeight": 35.5,
                                  "textAlign": "center",
                                },
                                undefined,
                              ]
                            }
                          >
                            Set up Kinly Overdraft
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenContainer>
      <View
        collapsable={false}
        pointerEvents="box-none"
        style={
          Object {
            "height": 44,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 1,
          }
        }
      />
    </View>
  </RNCSafeAreaProvider>
</View>
`;
